### PR TITLE
GUNSHIP/UAV/CRUISEMISSILE supports

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this?
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

Added back GUNSHIP,UAV,CRUISEMISSILE supports now all working

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [x] Yes, absolutely boyo

### How can the changes be tested?
Steps:
spawn support like you do
********************************************************
Notes:
Cruise missile currently targets and FOLLOWS its target, which is bad, possible solution, create some really small object, like energy battery(forgot classname for it so didn't do it myself) and attach laser target to it + I really wanted to launch 2 rockets, 1 HE and 1 Cluster, but reload time is abysmal, so possible solution is to create config fix for it and spawn second laucher.
UAV pretty much works as is, maybe some quirks to iron out
Gunship was a big pile of dog shit, figured it out eventually, there may be bugs (USAF gunship sucks at following orbit around target area) + default gunship routine should make use out of planes loadout (currently swaps pylons with rocket pods) + _oppositeSide should be added to make gunship target... well opposite side units properly(left comments so you can understand what I mean) + for nickel steell gunship I really wanted to make it "fire" it's search light at the target, added code for it but not tested it.

### CODE REVIEW

**Comment by wersal454** [comment]
after this is done ```+ default gunship routine should make use out of planes loadout (currently swaps pylons with rocket pods)``` gunships needs to be added to OPTRE and maybe to Star wars modsets or maybe test it with those to make sure it actually works and then push all together

**Comment by wersal454** [comment]
@Tiny-DM , @jaj22 
mentioning you just in case you guys want to younk some of that stuff and/or help out

**Comment by wersal454** [comment]
Almost forgot, there supposed to be a marker in gunship support to notify players what area gunship is targeting, couldn't figure it out so... hope one day, some one will

**Comment by wersal454** [comment]
Oh and don't forget to add gunships and attack uavs to airbase spawn pool

**Comment by wersal454** [comment]
Added small energy battery as C.M. target. After 20 seconds it will detach from the target so there will be some time to escape
Hopefully fixed issues with gunship behavior, kinda 1 problem remains, gunships won't target boats(seemingly) and planes(grounded) + still haven't done spotlight on nickel steel plane
Uav support I think works as is, though testing won't hurt

**Comment by wersal454** [comment]
Updated cruise missile support:
now it will fire cluster missile after HE one.
Updated Gunship support:
Now gunship will target boats/ships and grounded planes
Uav support works as is.

**Comment by wersal454** [comment]
Still need to add gunships and attack uav's to airbase spawn pool + gunships for templates like OPTRE

**Comment by wersal454** [comment]
Added gunships to OPTRE and star wars templates and USAF ac130 to RHS as optional mod.
Added attack uavs to airbase spawn pool so people can now use them.

File: A3A/addons/launchMissile.sqf
Comment (Silence-112) [inline A3A/addons/launchMissile.sqf:1]: I'm guessing this isn't meant to be here?

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Other than this rogue file, probably ready for testing

**Review by Silence-112 — COMMENTED** [review]
Ready for testing, probably

File: A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf [hunk 1]
```diff
@@ -318,6 +319,8 @@ if (!_busy) then {
                     + (_faction get "vehiclesPlanesCAS")
                     + (_faction get "vehiclesPlanesAA")
                     + (_faction get "vehiclesPlanesTransport");
+		    		+ (_faction get "vehiclesPlanesGunship");
```
Comment (Silence-112) [inline A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf]: This should probably be getOrDefault, incase the current template has no vehiclesPlanesGunship

File: A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf [hunk 2]
```diff
@@ -295,11 +295,12 @@ if (!_busy) then {
...
 	_groups pushBack _groupX;
 	_countX = 0;
-	while {_countX < 3} do {
+	private _vehCount = round (random [2, 4, 5]);
+	while {_countX < _vehCount} do {
 		private _veh = objNull;
 		private _spawnParameter = [_markerX, "Plane"] call A3A_fnc_findSpawnPosition;
 		if(_spawnParameter isEqualType []) then {
-			private _vehPool = (_faction get "vehiclesPlanesCAS") + (_faction get "vehiclesPlanesAA");
+			private _vehPool = (_faction get "vehiclesPlanesCAS") + (_faction get "vehiclesPlanesAA") + (_faction get "vehiclesPlanesGunship") + (_faction get "uavsAttack");
```
Comment (Silence-112) [inline A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf]: same here

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Double check each `get vehiclesPlanesGunship`, use getOrDefault where applicable

**Comment by wersal454** [comment]
Also, I probably forgot to check what is wrong with map marker

**Comment by wersal454** [comment]
> Also, I probably forgot to check what is wrong with map marker

image
this one

File: A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf (hunk 2 above)
Comment (wersal454) [reply A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf]: getOrDefault ["vehiclesPlanesGunship", "vehiclesPlanesCAS"]; and
getOrDefault ["uavsAttack", "vehiclesPlanesCAS"]; 
?

**Review by Silence-112 — APPROVED** [review]
lgtm

**Pull request merged**